lib/matplotlib/transforms.py

```diff
@@ -286,232 +281,216 @@ def is_unit(self):
-        the bounding box. It is not guaranteed to be sorted from left to
-        right.""")
+        the bounding box. It is not guaranteed to be sorted from left to right.
+        """
+        return self.get_points()[:, 0]
 
-    def _get_intervaly(self):
-        return self.get_points()[:, 1]
-    intervaly = property(_get_intervaly, None, None, """
+    @property
+    def intervaly(self):
+        """
         (property) :attr:`intervaly` is the pair of *y* coordinates that define
         the bounding box.  It is not guaranteed to be sorted from bottom to
-        top.""")
+        top.
+        """
+        return self.get_points()[:, 1]
 
-    def _get_width(self):
+    @property
+    def width(self):
+        """
+        (property) The width of the bounding box.  It may be negative if
+        :attr:`x1` < :attr:`x0`.
+        """
         points = self.get_points()
         return points[1, 0] - points[0, 0]
-    width = property(_get_width, None, None, """
-        (property) The width of the bounding box.  It may be negative if
-        :attr:`x1` < :attr:`x0`.""")
 
-    def _get_height(self):
+    @property
+    def height(self):
+        """
+        (property) The height of the bounding box.  It may be negative if
+        :attr:`y1` < :attr:`y0`.
+        """
         points = self.get_points()
         return points[1, 1] - points[0, 1]
-    height = property(_get_height, None, None, """
-        (property) The height of the bounding box.  It may be negative if
-        :attr:`y1` < :attr:`y0`.""")
 
-    def _get_size(self):
+    @property
+    def size(self):
+        """
+        (property) The width and height of the bounding box.  May be negative,
+        in the same way as :attr:`width` and :attr:`height`.
+        """
         points = self.get_points()
         return points[1] - points[0]
-    size = property(_get_size, None, None, """
-        (property) The width and height of the bounding box.  May be negative,
-        in the same way as :attr:`width` and :attr:`height`.""")
 
-    def _get_bounds(self):
+    @property
+    def bounds(self):
+        """
+        (property) Returns (:attr:`x0`, :attr:`y0`, :attr:`width`,
+        :attr:`height`).
+        """
         x0, y0, x1, y1 = self.get_points().flatten()
         return (x0, y0, x1 - x0, y1 - y0)
-    bounds = property(_get_bounds, None, None, """
-        (property) Returns (:attr:`x0`, :attr:`y0`, :attr:`width`,
-        :attr:`height`).""")
 
-    def _get_extents(self):
-        return self.get_points().flatten().copy()
-    extents = property(_get_extents, None, None, """
+    @property
+    def extents(self):
+        """
         (property) Returns (:attr:`x0`, :attr:`y0`, :attr:`x1`,
-        :attr:`y1`).""")
+        :attr:`y1`).
+        """
+        return self.get_points().flatten().copy()
 
     def get_points(self):
-        return NotImplementedError()
+        raise NotImplementedError
```

Comment:
https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement:
Otherwise, raise evaluates the first expression as the exception object. It must be either a subclass or an instance of BaseException. If it is a class, the exception instance will be obtained when needed by instantiating the class with no arguments.